crates/ntx-builder/src/state/mod.rs

```diff
@@ -178,7 +220,18 @@ impl State {
             } => {
                 self.add_transaction(id, nullifiers, network_notes, account_delta).await?;
             },
-            MempoolEvent::BlockCommitted { header: _, txs } => {
+            MempoolEvent::BlockCommitted { header, txs } => {
+                anyhow::ensure!(
+                    header.prev_block_commitment() == self.chain_tip.commitment(),
+                    "New block's parent commitment {} does not match local chain tip {}",
+                    header.prev_block_commitment(),
+                    self.chain_tip.commitment()
+                );
+
+                // Chain MMR always lags by one block.
+                self.chain_mmr.add_block(self.chain_tip.clone(), true);
+                self.chain_mmr.prune_to(..Self::CHAIN_TIP_LENGTH.into());
```

Comment:
@Mirko-von-Leipzig lmk if you would like to avoid pruning after every block add. For example we could have an initial length and then prune back to initial length once we reach 2x initial.

Reply:
I think pruning on every block is probably an overkill. We could also prune on every 8th block or something like that.

Separately, is `prune_to()` used correctly here? I though this would prune just the initial `CHAIN_TIP_LENGTH` from the MMR (while we want to prune to `chain_tip - CHAIN_TIP_LENGTH`, I think).

Reply:
Yea I wasn't sure whether `self.chain_mmr.chain_length()` returned block count or length of the chain. My understanding now is that it is the latter.